### DISCUSSION TITLE
Unable to generate images using S3

### DISCUSSION THREAD

**Discussion by morawcik** [spatie/laravel-medialibrary#3379] [Q&A]
Hi,
I'm trying to configure Media library with S3 but i can't. I have command to create products on my store (products imported from  my other source) and there I have simple image upload:
```$media = $product->addMediaFromUrl($imageUrl)->toMediaCollection('images');```
in .env:
I added ```MEDIA_DISK=s3``` to .env file and during upload it creates new folder on S3 bucket and upload there an original image but that's all - no other files or conversions. Also I'm getting errors like:

> Unable to read image from path (/tmp/GlideksSLqK)

or

> no decode delegate for this image format `' @ error/constitute.c/ReadImage/575 at ...

but with other MEDIA_DISK (like "public") it's working as charm without any problems.

**Comment by pratikbarman** [comment]
i am getting same issue.
